### PR TITLE
Calculator: Various improvements.

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

Sorry this PR took so long... working on other issues.

### Warning
  - **This PR requires the latest version of Math.js**

### Features
- Fixes a load of issues (See Section 2)
  - Fixes so correct answer shows when the user enters `0/0` == Error
  - Calculator was showing weird rounding issue
  - Now handling multiplications as space properly. ie `2^2 5` via UI
  - bails on spaces in query such as `5 232 + 234 * 2` as other search engines do
  - Calculator now handles modulo from search bar
- Now impossible to use the percentage sign as a modulo operator in the UI
- Stops showing hideous evaluated expression in subbar
- Handles queries such as `log of 5 plus 10 mult 8`

## Left todo

  - [x] Fix log/ln search bar issues
  - [x] Show BODMAS in search bar

## Related Issues and Discussions

Fixes: #4253
Fixes: #4279  
Fixes: #4166 
Fixes: #3933
Fixes: #3861 
Fixes: #1364 

## People to notify

@moollaza 
cc/ @mintsoft you might want to weigh in too

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/Calculator.pm [hunk 1]
```diff
@@ -10,16 +10,16 @@ use utf8;
...
+triggers query => $calc_regex;
 
-triggers query_nowhitespace => qr'^
-    (?: [0-9 () τ π e √ x × ∙ ⋅ * + \- ÷ / \^ \$ £ € \. \, _ ! = ]+ |
+triggers query => qr'^
+    (?: [0-9 () τ π e √ x × ∙ ⋅ * + \- ÷ / \^ \$ £ € \. \, _ ! = % ]+ |
     \d+\%=?$ |
-    what is| calculat(e|or) | solve | math |
-    times | divided by | plus | minus | cos | tau |
+    what\s?is| calculat(e|or) | solve | math |
```
Comment (mintsoft) [inline lib/DDG/Goodie/Calculator.pm]: do we really want `whatis` and `multby` or `dividedby` to fire?

**Comment by mintsoft** [comment]
@pjhampton I'm not so sure about the implicit multiplication on whitespace, I don't think I'd expect `1 234` to be parsed as `234`

**Comment by pjhampton** [comment]
@mintsoft if that is the case I have no idea how to solve #4166

**Comment by pjhampton** [comment]
1 234 shouldn't trigger in the first place

**Comment by mintsoft** [comment]
@pjhampton I mean in the context of a query like: `1 234 * 8` I would expect to get 9872, not 1872

**Comment by mintsoft** [comment]
In https://github.com/duckduckgo/zeroclickinfo-goodies/issues/4166 he's explicitly used the squared button, so that should always result (2^2) * 2, right?

**Comment by pjhampton** [comment]
I see, in the UI. I understand now. Thanks for clarifying @mintsoft

**Comment by moollaza** [comment]
Deployed to moollaza.duckduckgo.com with new Math.js 👍

File: share/goodie/calculator/calculator.js [hunk 2]
```diff
@@ -452,16 +453,8 @@ DDH.calculator = DDH.calculator || {};
...
-        if(/log|ln|!/.test(display.value)) {
-            normalizedExpression = normalizeExpression(display.value)
-        } else {
-            // helps with rounding errors
-            normalizedExpression = math.simplify(
-                normalizeExpression(display.value)
-            ).toString()                
-        }
+        var normalizedExpression = normalizeExpression(display.value);
+        console.log(normalizedExpression);
```
Comment (moollaza) [inline share/goodie/calculator/calculator.js]: Whoops, please remove!

**Comment by moollaza** [comment]
Deploying to Beta 👍

**Comment by moollaza** [comment]
@pjhampton Thanks! This LGTM :+1:

**Comment by bsstoner** [comment]
This was released to production 🚀

**Pull request merged**